src/main/java/seedu/address/model/person/insights/MostRecentEventInsight.java

```diff
@@ -0,0 +1,106 @@
+package seedu.address.model.person.insights;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.address.model.Model;
+import seedu.address.model.event.DateTime;
+import seedu.address.model.event.Event;
+import seedu.address.model.person.Person;
+
+/**
+ * This class encapsulates the insight of the most recent event.
+ */
+public class MostRecentEventInsight extends Insight implements Comparable<MostRecentEventInsight> {
+
+    // data field
+    private final DateTime dateTime;
+    private final boolean hasAtLeastOneEvent;
+
+    private MostRecentEventInsight(boolean hasAtLeastOneEvent) {
```

Comment:
Sorry, I don't really understand what you're trying to do with the constructors, do you mind explaining?

Reply:
Long story short, this is a dilemma we faced previously. I want child classes to have implemented methods, but they need to be static methods. My janky workaround for now is to have a dummy constructor to create a dummy object - made `private` for some semblance of safety - and then use that instance to call the overwritten method. 

A better implementation is probably to use a single `Insight` class with nested classes in it, but I wasn't sure.